msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/DiscussionForumServiceImpl.java

```diff
@@ -462,7 +468,18 @@ public ResourceProperties getEntityResourceProperties(Reference ref)
 
 	public String getEntityUrl(Reference ref)
 	{
-		// TODO Auto-generated method stub
+		if (StringUtils.isNotBlank(ref.getId())) {
+			String context = ref.getContext();
+			try{
+				DeveloperHelperService developerHelperService = (DeveloperHelperService) ComponentManager.get("org.sakaiproject.entitybroker.DeveloperHelperService");
```

Comment:
The line you suggested doesn't work. It causes a nullpointer exception because this is calling from the dashboard. Discussions is not the current placement. I searched in the code how this was resolved elsewhere and I found this solution that worked fine...

Reply:
Use the siteService, explore the pages and find the right placement, same strategy as developerHelperService does, better than injecting a dependency to the EB.

Reply:
Ok, I re-implemented the "getToolData" funcionality to remove the dependency to the EB.